mycroft/client/speech/service.py

```diff
@@ -125,66 +127,218 @@ def handle_speak(self, event):
         context = {'client_name': 'mycroft_listener',
                    'source': 'audio'}
         self.bus.emit(Message('speak', data, context))
 
-    def handle_sleep(self, event):
+    def handle_sleep(self, message):
         """Put the recognizer loop to sleep."""
         self.loop.sleep()
 
-    def handle_wake_up(self, event):
+    def handle_wake_up(self, message):
         """Wake up the the recognize loop."""
         self.loop.awaken()
 
-    def handle_mic_mute(self, event):
+    def handle_mic_mute(self, message):
         """Mute the listener system."""
         self.loop.mute()
 
-    def handle_mic_unmute(self, event):
+    def handle_mic_unmute(self, message):
         """Unmute the listener system."""
         self.loop.unmute()
 
-    def handle_mic_listen(self, _):
+    def handle_mic_listen(self, message):
         """Handler for mycroft.mic.listen.
 
         Starts listening as if wakeword was spoken.
         """
         self.loop.responsive_recognizer.trigger_listen()
 
-    def handle_mic_get_status(self, event):
+    def handle_mic_get_status(self, message):
         """Query microphone mute status."""
         data = {'muted': self.loop.is_muted()}
-        self.bus.emit(event.response(data))
+        self.bus.emit(message.response(data))
 
-    def handle_paired(self, event):
+    def handle_paired(self, message):
         """Update identity information with pairing data.
 
         This is done here to make sure it's only done in a single place.
         TODO: Is there a reason this isn't done directly in the pairing skill?
         """
-        IdentityManager.update(event.data)
+        IdentityManager.update(message.data)
 
-    def handle_audio_start(self, event):
+    def handle_audio_start(self, message):
         """Mute recognizer loop."""
         if self.config.get("listener").get("mute_during_output"):
             self.loop.mute()
 
-    def handle_audio_end(self, event):
+    def handle_audio_end(self, message):
         """Request unmute, if more sources have requested the mic to be muted
         it will remain muted.
         """
         if self.config.get("listener").get("mute_during_output"):
             self.loop.unmute()  # restore
 
-    def handle_stop(self, event):
+    def handle_stop(self, message):
         """Handler for mycroft.stop, i.e. button press."""
         self.loop.force_unmute()
 
+    # stt control bus events
+    def handle_load_stt_lang(self, message):
+        """ tell STT to pre load a model for this language, it will be needed """
+        lang = message.data.get("lang") or Configuration.get().get("lang", "en-us")
+        LOG.info(f"Loading STT lang: {lang}")
+        if hasattr(self.loop.stt, "load_language"):
+            try:
+                self.loop.stt.load_language(lang)
+            except:
+                LOG.exception(f"Failed to load STT lang: {lang}")
+        if hasattr(self.loop.fallback_stt, "load_language"):
```

Comment:
Maybe this PR is a good opportunity to refactor `fallback_stt` to a list of engines? If we're loading multiple languages, I think it's likely we'll have an engine per-language to maximize accuracy/performance

Reply:
answered this in other PR, but the vosk plugin is a great example of this,